docs/architecture/testing/test-framework.md

```diff
@@ -4,6 +4,36 @@ This document is an overview of the Mbed TLS test framework and test tools.
 
 This document is incomplete. You can help by expanding it.
 
+## Unit tests
+
+See https://tls.mbed.org/kb/development/test_suites
+
+### Unit test descriptions
+
+Each test case has a description which succinctly describes for a human audience what the test does. The first non-comment line of each paragraph in a `.data` file is the test description. The following rules and guidelines apply:
+
+* Test descriptions may not contain semicolons, line breaks and other control characters, or non-ASCII characters. <br>
+  Rationale: keep the tools that process test descriptions (`generate_test_code.py`, [outcome file](#outcome-file) tools) simple.
+* Test descriptions must be unique within a `.data` file. If you can't think of a better description, the convention is to append `#1`, `#2`, etc. <br>
```

Comment:
Will any error be raised if this is not conformed to?

Reply:
Yes for the “must”. But is it really necessary to document the tools in the architecture document?

Reply:
No, I was asking out of curiosity :)